### DISCUSSION TITLE
How can I add `bundleId` for java source context in sentry-react-native?

### DISCUSSION THREAD

**Discussion by lafest** [getsentry/sentry-react-native#3217] [Q&A]
I'm watching [this document](https://docs.sentry.io/platforms/android/enhance-errors/source-context/) and try to add java source context for my project.

In the document, sentry said that adding bundleId is required, like `options.addBundleId("A_VALID_UUID");`.

But, in the sentry-react-native repository, I could not find the option for adding bundleId. Is there any mistake for me or missing contents?

I already checked that java source context feature is added in sentry-java v6.22.0, and my project's sentry-react-native version(v5.7.1) is using upper sentry-java version.

**Comment by krystofwoldrich** [comment]
Hi, 
thank you for the question, this is correct. 

You are right, the `addBundleId` doesn't exist in React Native, you could use this option only when manually initializing the Android SDK.

But you can the other option and add the `bundleId` to this file [`sentry-debug-meta.properties`](https://docs.sentry.io/platforms/android/enhance-errors/source-context/#sentry-debug-metaproperties).

**Reply by lafest** [reply]
thanks for answer.

> But you can the other option and add the bundleId to this file [sentry-debug-meta.properties](https://docs.sentry.io/platforms/android/enhance-errors/source-context/#sentry-debug-metaproperties).

does this mean that I can use java source context feature in react native project by using `sentry-debug-meta.properties` file?

**Reply by lafest** [reply]
Additionally, is there any plan for supporting java source context for react native? (ex. add bundleId option for init function, add `addBundleId` method)

**Reply by lafest** [reply]
@krystofwoldrich hi, can you answer about additional questions?

**Reply by krystofwoldrich** [reply]
Hi, sorry for the late response time, 
yes by using `sentry-debug-meta.properties` you could use Java context feature, but note we have not tested this in RN app yet.

We likely won't add the `bundleId` option at the moment, as we would like to support the automatic source upload https://docs.sentry.io/platforms/android/enhance-errors/source-context/#using-the-gradle-build-tool-plugin
